massa-grpc/src/stream/subscribe_new_operations.rs

```diff
@@ -0,0 +1,169 @@
+use crate::error::GrpcError;
+use crate::service::MassaGrpcService;
+use futures_util::StreamExt;
+use massa_models::operation::{OperationType, SecureShareOperation};
+use massa_proto::massa::api::v1::{
+    self as grpc, BytesMapFieldEntry, OperationStreamFilterType,
+    SubscribeNewOperationsStreamRequest, SubscribeNewOperationsStreamResponse,
+};
+use std::pin::Pin;
+use tokio::select;
+use tonic::codegen::futures_core;
+use tonic::{Request, Streaming};
+use tracing::log::error;
+
+/// type declaration for StreamTransactionsThroughputStream
+pub type SubscribeNewOperationsStream = Pin<
+    Box<
+        dyn futures_core::Stream<Item = Result<SubscribeNewOperationsStreamResponse, tonic::Status>>
+            + Send
+            + 'static,
+    >,
+>;
+
+pub(crate) async fn subscribe_new_operations(
+    grpc: &MassaGrpcService,
+    request: Request<Streaming<SubscribeNewOperationsStreamRequest>>,
+) -> Result<SubscribeNewOperationsStream, GrpcError> {
+    let (tx, rx) = tokio::sync::mpsc::channel(grpc.grpc_config.max_channel_size);
+    let mut in_stream = request.into_inner();
+    let mut subscriber = grpc.pool_channels.operation_sender.subscribe();
+
+    tokio::spawn(async move {
+        if let Some(Ok(request)) = in_stream.next().await {
+            let mut request_id = request.id;
+            let mut filter = request.filter;
```

Comment:
wrap the filter with xxQuery object to match the schema convention(autotgenerated)